Add bs-node-config

seprich opened the pull request:
This library (https://www.npmjs.com/package/bs-node-config) is similar to node-config (http://lorenwest.github.io/node-config/) , config library written for node, but bs-node-config is written completely in ReasonML (doesn't use or depend on node-config). Features are similar but partially different too, e.g. strong validation of config parameters and json parsing of environment variable values are included. Node is the only viable platform because depends on nodejs fs api.

glennsl commented:
Thanks! This looks really good, but I'm a bit hesitant about the name. The `bs-` convention is usually (but definitely not always) used for bindings, and `bs-` followed by an existing npm package name strongly suggests that these are bindings to that package. I understand that the API is heavily inspired by the existing library, and that you therefore want to include that in the name, so perhaps `reason-node-config` or `re-node-config` might be better. Or something more clever. When I reimplemented `debug` in Reason I called it `rebug`.

You should also namespace the package, especially when the name is so generic. If someone decides to make bindings to `node-config` and follow the existing convention of calling it `bs-node-config`, even if _they_ namespace it, it's going to be confused with this, which is unfortunate for everyone. You are likely to get bug reports intended for them, and vice versa.

I'm not going to block you from redex over this, but I would strongly recommend a name change and will await merging until you've considered it at least.

seprich commented:
Hi! Thanks for the recommendations. Those were really good points. As a result I scoped the package under "@veikkaus/" scope and changed base name to "re-node-config". This pull request now updated too.
